src/VisualStudio/IntegrationTest/TestUtilities/InProcess/FindReferencesWindow_InProc.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using EnvDTE80;
+using Microsoft.VisualStudio.IntegrationTest.Utilities.Common;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.TableControl;
+using Microsoft.VisualStudio.Shell.TableManager;
+
+namespace Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess
+{
+    internal class FindReferencesWindow_InProc : InProcComponent
+    {
+        public static FindReferencesWindow_InProc Create() => new FindReferencesWindow_InProc();
+
+        public Reference[] GetContents(string windowCaption)
+        {
+            return InvokeOnUIThread(() =>
```

Comment:
At least *some* of it is required. The DTE calls would be marshaled to the UI thread regardless. _Maybe_ we could access the table control from a background thread, but there's no benefit to adding that complexity.

We already have a cross-process communication mechanism; I see no need to throw a cross-process COM call in there as well. :-)